sympy/polys/rootoftools.py

```diff
@@ -772,15 +775,94 @@ def _get_roots(cls, method, poly, radicals):
         free_names = {str(i) for i in poly.free_symbols}
         for x in chain((symbols('x'),), numbered_symbols('x')):
             if x.name not in free_names:
+                # xreplace makes alg field domain back into EX
+                # XXX: so make sure it stays same?
                 poly = poly.xreplace({d: x})
+                if dom.is_AlgebraicField:
+                    poly = Poly(poly, domain=dom)
```

Comment:
Why does this need to change the symbols at all?

Is it possible to just remove this whole section or does that break something?

Reply:
I looked into this. Removing it breaks `test_issue_19113()` in test_rootoftools. This led me to #19113 and #19117 -- I think basically this symbol change is done to always return RootOf objects with canonical `x` generator.

Reply:
Changing the `xreplace` to a `sub` preserves the domain, i.e. all tests pass with `sub` and removing lines 781-782 above. But, not sure if there are any other differences between `xreplace` and `sub` we should be careful for.

Reply:
There should be a better way than using `xreplace` that preserves the domain. It seems that `replace` can do it: `poly.replace(d, x)`.

Reply:
The subs method just calls replace.

Reply:
Sounds good, thanks

Reply:
See also gh-26808